packages/components/codemods/utils/transformV1ButtonPropsToButtonOrLinkButton.ts

```diff
@@ -0,0 +1,207 @@
+import ts from 'typescript'
+import { type ButtonProps as V1ButtonProps } from '~components/Button'
+import { type ButtonProps as NextButtonProps } from '~components/__next__/Button'
+
+import {
+  createJsxElementWithChildren,
+  createProp,
+  createStringProp,
+  getPropValueText,
+} from '../utils'
+
+export const migrateV1ButtonSize = (
+  oldValue: Exclude<V1ButtonProps['size'], undefined>,
+): Exclude<NextButtonProps['size'], undefined> => {
+  switch (oldValue) {
+    case 'small':
+      return 'medium'
+    case 'regular':
+      return 'large'
+  }
```

Comment:
Should we handle undefined here? Probably better off not using a switch too.

```suggestion
  const buttonSizeMap = {
    'small': 'medium',
    'regular': 'large',
  };

  return buttonSizeMap[oldValue] || undefined;
```

Reply:
Given the simplicity of this util and condition I've opted to just handle this in the`transformButtonProp` function within the `size` case. Keen to hear thoughts :)